### PR TITLE
semantic_text: avoid nesting for multiple levels

### PR DESCRIPTION
**Pull request by carlosdelest**
Follow up to https://github.com/elastic/elasticsearch/pull/110055, addresses multiple nesting levels when using objects as nested fields

### CODE REVIEW

**Comment by carlosdelest** [comment]
@elasticsearchmachine run elasticsearch-ci/part-2

File: server/src/main/java/org/elasticsearch/index/mapper/MapperBuilderContext.java [hunk 1]
```diff
@@ -135,7 +139,10 @@ public MergeReason getMergeReason() {
         return mergeReason;
     }
 
-    public boolean isNested() {
```
Comment (carlosdelest) [inline server/src/main/java/org/elasticsearch/index/mapper/MapperBuilderContext.java:138]: Renamed to make it more clear that it belongs to a nested context

File: server/src/main/java/org/elasticsearch/index/mapper/NestedObjectMapper.java [hunk 2]
```diff
@@ -182,11 +182,6 @@ public MapperBuilderContext createChildContext(String name, Dynamic dynamic) {
                 getMergeReason()
             );
         }
-
```
Comment (carlosdelest) [inline server/src/main/java/org/elasticsearch/index/mapper/NestedObjectMapper.java:185]: No need to override as it is taken care of via the superclass constructor / attribute

File: server/src/main/java/org/elasticsearch/index/mapper/MapperBuilderContext.java [hunk 3]
```diff
@@ -135,7 +139,10 @@ public MergeReason getMergeReason() {
...
-        return false;
+    /**
+     * Returns true if this context is included in a nested context, either directly or any of its ancestors.
```
Comment (Mikep86) [inline server/src/main/java/org/elasticsearch/index/mapper/MapperBuilderContext.java:143]:
```suggestion
     * Returns true if this context is included in a nested context, either directly or via any of its ancestors.
```

File: server/src/test/java/org/elasticsearch/index/mapper/MapperBuilderContextTests.java [hunk 4]
```diff
@@ -27,4 +27,11 @@ public void testRootWithMergeReason() {
         assertEquals(mergeReason, root.getMergeReason());
     }
 
+    public void tesIsInNestedContext() {
+        MapperBuilderContext root = MapperBuilderContext.root(true, false);
```
Comment (Mikep86) [inline server/src/test/java/org/elasticsearch/index/mapper/MapperBuilderContextTests.java:31]: It doesn't affect the test, but why did you set `isSourceSynthetic` to `true` here? It doesn't seem to have any impact on the tested functionality.

**Review by Mikep86 — APPROVED** [review]
LGTM!

File: server/src/test/java/org/elasticsearch/index/mapper/MapperBuilderContextTests.java (hunk 4 above)
Comment (carlosdelest) [reply server/src/test/java/org/elasticsearch/index/mapper/MapperBuilderContextTests.java:31]: Correct, I just filled in with some data. I could use random values but didn't think it worthwhile, as it won't impact the response.

Comment (Mikep86) [reply server/src/test/java/org/elasticsearch/index/mapper/MapperBuilderContextTests.java:31]: Agreed, random values aren't worth it here. 99% of times I've used `MapperBuilderContext.root`, both args were set to `false`, so IMO that's good enough for here too.

**Pull request merged**